testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -162,6 +162,25 @@ def test_AnalysisFromFunction(u, start, stop, step, nframes):
         assert_equal(results, ana.results)
 
 
+def test_AnalysisFromFunction_args_order(u):
+    # Issue 2503
+    def mass_xyz(atomgroup, masses):
+        return atomgroup.positions * masses
+    
+    protein = u.select_atoms('protein')
+    sub = u.atoms[:100]
+    #ans1 = base.AnalysisFromFunction(mass_xyz, u.trajectory, protein, protein.masses.reshape(-1, 1))
+    # lack traj but can find in protein
+    ans2 = base.AnalysisFromFunction(mass_xyz, protein, protein.masses.reshape(-1, 1))
+    # multiple traj input
+    ans3 = base.AnalysisFromFunction(mass_xyz, protein, sub, protein.masses.reshape(-1, 1))
```

Comment:
Third test.

Reply:
Also, just a note, `sub.universe.trajectory is protein.universe.trajectory` i.e. the [trajectories are the same object in memory](http://net-informations.com/python/iq/is.htm), so your test for multiple trajectory inputs would always succeed but for the wrong reasons. To test that you are always finding the *first* trajectory, you would need to create different universes and check `ans._trajectory is protein.universe.trajectory`.